lib/bootstrapper/index.ts

```diff
@@ -28,17 +27,18 @@ export class BootstrapPgStac extends Construct {
   constructor(scope: Construct, id: string, props: BootstrapPgStacProps) {
     super(scope, id);
 
-    const { pgstacVersion = DEFAULT_PGSTAC_VERSION } = props;
     const handler = new aws_lambda.Function(this, "lambda", {
-      handler: "handler.handler",
-      runtime: aws_lambda.Runtime.PYTHON_3_8,
-      code: aws_lambda.Code.fromDockerBuild(__dirname, {
+      ...props.lambdaFunctionOptions ?? {
```

Comment:
Maybe someone with more TS experience than me can weigh in here (@alukach ?).  Is completely selecting either a fully specified `CustomLambdaFunctionOptions` interface or instead the manually specified "default" parameters the best approach for doing this?  

I would have guessed that using some combination of an interface with defaults and destructuring was more elegant but it has been a long time since I wrote any TS :]

Reply:
I think I would agree that this is possibly less ideal for two reasons:

1. It only allows users to specify _all_ options or partial.
2. We're using our own custom interface (`CustomLambdaFunctionOptions`) which isn't based on the actual [`aws_lambda.FunctionProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html) that CDK uses. It's a subset of all the options a user may want to set and we will have to keep in sync with the lambda props as they change in future CDK versions.

I think a way around both of these would be for us to instead allow a user to submit none, some, or all the types specified in `aws_lambda.FunctionProps` and then apply defaults for properties that are not specified by the user.  I think we can achieve this via JS' destructuring (and the neat fact that you can destructure `undefined` without error) and TS' [`Partial` & `Omit` utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html):

```ts

export class BootstrapPgStac extends Construct {
  // ...

  constructor(scope: Construct, id: string, props: BootstrapPgStacProps) {
    // ...

    const handler = new aws_lambda.Function(this, "lambda", {
      // Defaults for configurable properties
      runtime: aws_lambda.Runtime.PYTHON_3_8,
      handler: "handler.handler",
      memorySize: 128,
      logRetention: aws_logs.RetentionDays.ONE_WEEK,
      timeout: Duration.minutes(2),

      // Configurable properties will overwrite defaults
      ...props.lambdaFunctionOptions,
      
      // Non-configurable
      code: props.lambdaAssetCode ?? aws_lambda.Code.fromDockerBuild(__dirname, {
        file: "runtime/Dockerfile",
      }),
      vpc: hasVpc(props.database) ? props.database.vpc : props.vpc,
    });
    
    // ...
  }
}


export interface BootstrapPgStacProps {
  //...

  // Allow user to specify none, some, or all props in AWS Lambda Function Props,
  // except for `code`.
  readonly lambdaFunctionOptions?: Partial<Omit<aws_lambda.FunctionProps, "code" | "vpc">>;

  //...
}

```

Reply:
Thinking more about this, we can probably do away with the `lambdaAssetCode` and instead not omit the `code` property from `aws_lambda.FunctionProps`.

Reply:
@emileten Is this something you want to try and tackle solo or should we hop in and try to tag team some revisions.

Reply:
Just looked into this in detail (was off on monday), I totally agree that : 

> It only allows users to specify all options or none.
>  We're using our own custom interface

Is not good, I just had no way around this, but @alukach guidance sounds great, trying to implement this.

Reply:
Unfortunately, I think we won't be able to do that because those are not supported by JSII @alukach, see https://github.com/aws/jsii/issues/2901. 

My suggested workaround is to say that the user can pass anything from the aws_lambda.FunctionProps type but raise an error if it's vpc/subnet/role etc..., and say it in the docs. It's imperfect because e.g. the user will not be warned by the type checker, but better than copy/pasting properties which is what i was doing initially.

Reply:
side note, looks like the repo this issue originated from (maintained by aws) serves the same purpose as ours -- defining shareable constructs --  but for typical multi-aws-service deployments. Could be a source of inspiration !

https://github.com/awslabs/aws-solutions-constructs

Reply:
see https://github.com/developmentseed/eoapi-cdk/pull/74/commits/dbcada8cf65e3264a71144c660e9a688db9f1e59